src/context/AuthContext.js

```diff
@@ -0,0 +1,117 @@
+import axios from "axios";
+import { createContext, useContext, useReducer, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { authReducer } from "../reducer/AuthReducer";
+
+const AuthContext = createContext(null);
+
+const initialState = {
+	firstName: "",
+	email: "",
+	password: "",
+	error: false,
+	userData: "",
+	token: "",
+};
+
+const AuthProvider = ({ children }) => {
+	const [authState, authDispatch] = useReducer(authReducer, initialState);
+	const { firstName, email, password, userData, token, error } = authState;
+	const navigate = useNavigate();
+
+	const logIn = async () => {
+		try {
+			const response = await axios.post("/api/auth/login", {
+				email,
+				password,
+			});
+			if (response.status === 200) {
+				localStorage.setItem("token", response.data.encodedToken);
+				localStorage.setItem(
+					"userData",
+					JSON.stringify(response.data.foundUser)
+				);
+
+				authDispatch({
+					type: "LOG-IN",
+					payload: {
+						userData: response.data.foundUser,
+						token: response.data.encodedToken,
+					},
+				});
+				navigate("/product");
+			}
+		} catch (err) {
+			authDispatch({
+				type: "ERROR",
+			});
+		}
+	};
+
+	const signUp = async () => {
+		try {
+			const response = await axios.post("/api/auth/signup", {
+				firstName,
+				email,
+				password,
+			});
+			if (response.status === 200 || 201) {
+				localStorage.setItem("token", response.data.encodedToken);
+				localStorage.setItem(
+					"userData",
+					JSON.stringify(response.data.createdUser)
+				);
+				authDispatch({
+					type: "SIGN-UP",
+					payload: {
+						userData: response.data.createdUser,
+						token: response.data.encodedToken,
+					},
+				});
+				navigate("/");
+			}
+		} catch (err) {
+			authDispatch({ type: "ERROR" });
+		}
+	};
+
+	const signOut = () => {
+		authDispatch({ type: "TOKEN-REMOVED" });
```

Comment:
This type could be named better. As we know that we are removing the token on signout but it doesn't make it clear in this dispatch. You could name it something as `LOGOUT`